src/lib.rs

```diff
@@ -277,7 +275,16 @@ impl Metadata {
     /// ```
     pub fn new_from_path<S: AsRef<ffi::OsStr>>(path: S) -> Result<Metadata> {
         let mut err: *mut gexiv2::GError = ptr::null_mut();
-        let c_str_path = ffi::CString::new(path.as_ref().as_bytes()).unwrap();
+
+        // let c_str_path = ffi::CString::new(path.as_ref().as_bytes()).unwrap();
+        let c_str_path = ffi::CString::new(
+            path
+            .as_ref()
+            .to_str()
+            .unwrap()
+            .as_bytes()
```

Comment:
As above, this will be grabbing the bytes of a UTF-8 encoded Rust string... but on Windows we would need UTF-16.

I tried to look into how the Rust standard library handles path-specific stuff. What I was _hoping_ to find was some kind of generic helper function like (hypothetically) `to_native_platform_path_string_pointer` (maybe with a less clunky name :sweat_smile:). Instead, though, it looks like there are platform-specific implementations with totally different behaviour.

The UNIX-like ones rely on the `std::os::unix::ffi::OsStrExt` `as_bytes()` method, which is at least straightforward.

The Windows ones however were a bit harder to understand. To investigate how it works, I picked a random filesystem operation on paths: `symlink_file` (https://doc.rust-lang.org/src/std/os/windows/fs.rs.html#564-566) which relies on something mysterious called `symlink_inner` which doesn't show up in the online docs. Looking at the source, though, it's [here](https://github.com/rust-lang/rust/blob/8c41305631f96bae3e456d7942f00e0d437c350e/library/std/src/sys/windows/fs.rs#L1172).

That function actually takes two paths, and seems to use _two different ways_ to convert them to pointers to pass to Windows. I don't know why there are two different mechanisms or what the difference is between them, but they are `to_u16s` (https://github.com/rust-lang/rust/blob/8c41305631f96bae3e456d7942f00e0d437c350e/library/std/src/sys/windows/mod.rs#L162) and `maybe_verbatim` (https://github.com/rust-lang/rust/blob/8c41305631f96bae3e456d7942f00e0d437c350e/library/std/src/sys/windows/path.rs#L229).

I don't know yet where to go from here. I still kind of wish there was a generic `to_platform_path_ptr()` or whatever which could hide/abstract away the internal OS-specific implementations. I don't know if there's a good reason that doesn't exist. Perhaps that's something we can write a local helper for?